android/src/com/mapswithme/util/bottomsheet/MenuBottomSheetFragment.java

```diff
@@ -0,0 +1,106 @@
+package com.mapswithme.util.bottomsheet;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.FragmentManager;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
+import com.mapswithme.maps.R;
+
+import java.util.ArrayList;
+
+public class MenuBottomSheetFragment extends BottomSheetDialogFragment
+{
+
+    private final Builder builder;
+
+    MenuBottomSheetFragment(Builder builder)
+    {
+        this.builder = builder;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState)
+    {
+        return inflater.inflate(R.layout.bottom_sheet, container);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState)
+    {
+        super.onViewCreated(view, savedInstanceState);
+        TextView titleView = view.findViewById(R.id.bottomSheetTitle);
+        RecyclerView recyclerView = view.findViewById(R.id.bottomSheetMenuContainer);
+        titleView.setText(builder.getTitle());
+
+        MenuAdapter menuAdapter = new MenuAdapter(builder.getMenuBottomSheetItems(), this::onMenuItemClick);
+        recyclerView.setAdapter(menuAdapter);
+        recyclerView.setLayoutManager(new LinearLayoutManager(requireActivity()));
+    }
+
+    private void onMenuItemClick(MenuBottomSheetItem item)
+    {
+        dismiss();
+    }
+
+    public static class Builder
```

Comment:
Is this Builder wrapper really necessary? Can we use YAGNI + KISS here?

Reply:
Again got inspiration from other libraries. But yes I should be able to provide all necessary parameters in the constructor so this builder should not be needed.